br/pkg/lightning/backend/local/local.go

```diff
@@ -1145,14 +1145,16 @@ func (local *Backend) generateAndSendJob(
 	for _, jobRange := range jobRanges {
 		r := jobRange
 		eg.Go(func() error {
-			select {
-			case <-egCtx.Done():
+			if egCtx.Err() != nil {
 				return nil
-			default:
 			}
 
+			failpoint.Inject("beforeGenerateJob", nil)
 			jobs, err := local.generateJobForRange(egCtx, engine, r, regionSplitSize, regionSplitKeys)
 			if err != nil {
+				if common.IsContextCanceledError(err) {
+					return nil
```

Comment:
if we return nil here, caller will continue running

Reply:
it's expected? because context is `egCtx` and in line 1148 they will fail fast.

Reply:
will this happen:
- parent ctx cancelled,`generateAndSendJob` returns nil
- all works returns nil on ctx.Done()
- then we log that engine import success

maybe add ctx.Err into firstErr before return

Reply:
https://github.com/pingcap/tidb/pull/44734/commits/2d3ca6b1585ef9ca2de4770e9fddb3e290f39df4